marketplace/mainTemplate.json

```diff
@@ -0,0 +1,223 @@
+          "type": "Bool",
+          "defaultValue": true,
+          "metadata": {
+              "description": "enable auto scaling"
+          }
+      },
+      "vmCount": {
+          "type": "Int",
+          "defaultValue": 3,
+          "metadata": {
+              "description": "VM count"
+          }
+      },
+      "dnsPrefix": {
+          "defaultValue": "[concat(parameters('clusterResourceName'),'-dns')]",
+          "type": "String",
+          "metadata": {
+              "description": "Optional DNS prefix to use with hosted Kubernetes API server FQDN."
+          }
+      },
+      "osDiskSizeGB": {
+          "defaultValue": 0,
+          "minValue": 0,
+          "maxValue": 1023,
+          "type": "Int",
+          "metadata": {
+              "description": "Disk size (in GiB) to provision for each of the agent pool nodes. This value ranges from 0 to 1023. Specifying 0 will apply the default disk size for that agentVMSize."
+          }
+      },
+      "kubernetesVersion": {
+          "type": "String",
+          "defaultValue": "1.28.9",
+          "metadata": {
+              "description": "The version of Kubernetes."
+          }
+      },
+      "networkPlugin": {
+          "defaultValue": "kubenet",
+          "allowedValues": [
+              "azure",
+              "kubenet"
+          ],
+          "type": "String",
+          "metadata": {
+              "description": "Network plugin used for building Kubernetes network."
+          }
+      },
+      "enableRBAC": {
+          "defaultValue": true,
+          "type": "Bool",
+          "metadata": {
+              "description": "Boolean flag to turn on and off of RBAC."
+          }
+      },
+      "enablePrivateCluster": {
+          "defaultValue": false,
+          "type": "Bool",
+          "metadata": {
+              "description": "Enable private network access to the Kubernetes cluster."
+          }
+      },
+      "enableHttpApplicationRouting": {
+          "defaultValue": true,
+          "type": "Bool",
+          "metadata": {
+              "description": "Boolean flag to turn on and off http application routing."
+          }
+      },
+      "enableAzurePolicy": {
+          "defaultValue": false,
+          "type": "Bool",
+          "metadata": {
+              "description": "Boolean flag to turn on and off Azure Policy addon."
+          }
+      },
+      "enableSecretStoreCSIDriver": {
+          "defaultValue": false,
+          "type": "Bool",
+          "metadata": {
+              "description": "Boolean flag to turn on and off secret store CSI driver."
+          }
+      },
+      "osSKU": {
+          "type": "string",
+          "defaultValue": "AzureLinux",
+          "allowedValues": [
+              "AzureLinux",
+              "Ubuntu"
+          ],
+          "metadata": {
+              "description": "The Linux SKU to use."
+          }
+      }
+  },
+  "variables": {
+      "plan-name": "DONOTMODIFY",
```

Comment:
I... assume... we shouldn't modify these?  Something behind the scenes in the process will do what's needed with these values?

To vet this template I tried a deploy (knowing that the custom spinkube part would fail) but it failed first on this bit:
```
$ az deployment group create   --name vdice-test-deployment --resource-group vdice --template-file mainTemplate.json
Please provide string value for 'extensionResourceName' (? for help): spinkube
Please provide string value for 'clusterResourceName' (? for help): vdice-aks
Please provide string value for 'location' (? for help): ?
The location of AKS resource.
Please provide string value for 'location' (? for help): westus2
{"status":"Failed","error":{"code":"DeploymentFailed","target":"/subscriptions/54555704-ba9a-4bc8-959b-91fdb3ac2af1/resourceGroups/vdice/providers/Microsoft.Resources/deployments/vdice-test-deployment","message":"At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-deployment-operations for usage details.","details":[{"code":"ExtensionTypeRegistrationGetFailed","message":"Extension type 'DONOTMODIFY' is not supported in region 'westus2' for cluster type: ManagedClusters. Please refer to https://aka.ms/k8s-extensions-list-cluster for a list of supported extension types for this cluster.","additionalInfo":[]}]}}
```